api/src/test/java/org/openmrs/module/fgh/mpi/MpiIntegrationProcessorTest.java

```diff
@@ -371,4 +426,94 @@ public void process_shouldProcessAPersonDeleteEvent() throws Exception {
 	public void process_shouldProcessAPatientDeleteEvent() throws Exception {
 	}
 	
+	@Test
+	public void process_shouldInitTheContextWithoutSSL() throws Exception {
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		Mockito.verify(snapshotEventProcessor, times(0))
+		        .process(new DatabaseEvent(1234, "patient", UPDATE, null, null, null));
+	}
+	
+	@Test
+	public void process_shouldInitTheContextWithtSSL() throws Exception {
+		AuthenticationType SSL = AuthenticationType.SSL;
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(SSL.toString());
+		Mockito.verify(snapshotEventProcessor, times(0))
+		        .process(new DatabaseEvent(12345, "patient", UPDATE, null, null, null));
+	}
+	
+	@Test
+	public void process_shouldgetPatientInRemoteMasterPatientIndexAsSanteMPI() throws Exception {
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		final String patientUuid = "patient-uuid-for-sante";
+		Map res = new HashMap();
+		res.put("active", true);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(res);
+		Map prevState = singletonMap("uuid", patientUuid);
+		assertNotNull(processor.process(1, new DatabaseEvent(null, "person", DELETE, null, prevState, null)));
+		Mockito.verify(mockMpiHttpClient, times(1)).getPatient(patientUuid);
+	}
+	
+	@Test
+	public void process_shouldgetPatientInRemoteMasterPatientIndexAsOpenCR() throws Exception {
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM_AS_OPENCR.toString());
+		final String patientUuid = "patient-uuid-for-openCR";
+		Map res = new HashMap();
+		res.put("active", true);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(res);
+		Map prevState = singletonMap("uuid", patientUuid);
+		assertNotNull(processor.process(1, new DatabaseEvent(null, "person", DELETE, null, prevState, null)));
+		Mockito.verify(mockMpiHttpClient, times(1)).getPatient(patientUuid);
+	}
+	
+	@Test
+	public void shouldProcessBundleData() throws Exception {
```

Comment:
What method are you testing? Please make sure you call it in the test?

Reply:
I added those tests in the SnapshotEventProcessorTest but if you want i can create a separete file for them